quickwit-metastore/src/split_metadata_version.rs

```diff
@@ -138,6 +143,7 @@ pub(crate) struct SplitMetadataV1 {
 impl From<SplitMetadataV1> for SplitMetadata {
     fn from(v1: SplitMetadataV1) -> Self {
         let source_id = v1.source_id.unwrap_or_else(|| "unknown".to_string());
+        let index_id = v1.index_id.unwrap_or_else(|| "unknown".to_string());
```

Comment:
Remove this line and use `index_id: v1.index_id` below. That way, we can test for emptiness when overriding the index ID.